### PR TITLE
Fixes scars sticking around after a limb is dismembered

### LINKED ISSUE #53729: You can see scars on body parts that are not connected to the body
You can see scars on body parts that are not connected to the body
image

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
During dismemberment from wounds, some scars would stick around even though the limb was no longer attached to the person. This was because I removed the scars from the person before removing the wounds, which itself would create more scars that would still be tied to the victim's scars. This fixes that, and tidies up some related code as well. Notably, you can now see which types of wounds a detached limb is suffering from by sight. Also, I have no clue what happened before when you attached a limb with existing wounds to someone, but I've made sure it applies the wounds to the person where they left off.

[](https://i.imgur.com/BNHzmeI.png)

I also realized I made an order of operations error in the self-treatment for broken bones (with the bone gel and surgical tape) on the bone gel step, which made failing these treatments and briefly passing out/wasting a use of the gel 38 percentage points more likely than intended. This corrects that, so the base is 25%/45% failure for severe/critical blunt wounds. It also raises the bonuses to succeeding you get from painkillers a bit since you're going through a lot of trouble for all this already.

Fixes: #53729

## Why It's Good For The Game
Scars are more sensible, self treating bones is less impossible (but still hard)

## Changelog
:cl: Ryll/Shaps
fix: Dismemberment will no longer lead to phantom scars showing up on a person's missing limbs. Insert your own MGSV joke here.
fix: Fixed some bad logic that made self-treating severe/critical broken bones with bone gel have 63%/83% base fail rates, instead of 25%/45% as intended. As such, self-treating these wounds is now far less likely to fail, though you'll still have to wait for the effect.
tweak: The bonuses to the above self-treatment success rates from drunkenness, morphine, and determination are a bit higher as well.
tweak: You can now see what wound types a detached limb is suffering by inspecting it
/:cl:

### CODE REVIEW

**Comment by Couls** [comment]
Glad to see you're still maintaining wounds

File: code/modules/surgery/bodyparts/dismemberment.dm [hunk 1]
```diff
@@ -359,15 +358,17 @@
 	for(var/obj/item/organ/O in contents)
 		O.Insert(C)
 
+	for(var/i in wounds)
+		var/datum/wound/W = i
+		W.apply_wound(src, TRUE)
+
 	for(var/thing in scars)
 		var/datum/scar/S = thing
+		if((S in C.all_scars))
```
Comment (Rohesie) [inline code/modules/surgery/bodyparts/dismemberment.dm:367]:
```suggestion
		if(S in C.all_scars)
```

File: code/modules/surgery/bodyparts/dismemberment.dm [hunk 2]
```diff
@@ -359,15 +358,17 @@
 	for(var/obj/item/organ/O in contents)
 		O.Insert(C)
 
+	for(var/i in wounds)
+		var/datum/wound/W = i
+		W.apply_wound(src, TRUE)
```
Comment (Rohesie) [inline code/modules/surgery/bodyparts/dismemberment.dm:363]: I'm a bit confused by this logic.
This copies the limb's list of wounds first.
Then it goes to every wound and tells them to re-apply themselves to the limb.
Aren't they caught here and delete themselves, in `apply_wound()`?
```cpp
	for(var/i in L.wounds)
		var/datum/wound/preexisting_wound = i
		if((preexisting_wound.type == type) && (preexisting_wound != old_wound))
			qdel(src)
			return
```

**Comment by Ryll-Ryll** [comment]
@Rohesie should be good, included a few other fixes and tweaks as well as mentioned in the OP and changelog

**Pull request merged**